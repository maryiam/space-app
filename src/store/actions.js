export const FIND_PAYMENT = {
  REQUEST: 'payment/FIND/REQUEST',
  SUCCESS: 'payment/FIND/SUCCESS',
  FAILURE: 'payment/FIND/FAILURE'
};

export const FIND_DESTINATION = {
  SUCCESS: 'destination/FIND/SUCCESS',
  FAILURE: 'destination/FIND/FAILURE'
};

export const SET_SEARCH = 'destination/SET_SEARCH';
export const SET_DESTINATION = 'order/SET_DESTINATION';
export const RESET_DESTINATION = 'order/RESET_DESTINATION';
export const SET_PAYMENT = 'order/SET_PAYMENT';
export const SUBMIT = 'order/SUBMIT';

export function setPayment(paymentId) {
  return {
    type: SET_PAYMENT,
    paymentId
  };
}

export function startTrip() {
  return {
    type: SUBMIT
  };
}

export function setSearch(searchCriteria) {
  return {
    type: SET_SEARCH,
    searchCriteria
  };
}

export function setDestination(destinationId) {
  return {
    type: SET_DESTINATION,
    destinationId
  };
}

export function resetDestination() {
  return {
    type: RESET_DESTINATION
  };
}
