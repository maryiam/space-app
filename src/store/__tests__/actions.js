import * as actions from '../actions';

describe('store/actions', () => {
  describe('#setPayment', () => {
    it('creates the correct action', () => {
      const action = actions.setPayment(0);
      expect(action).toMatchSnapshot();
    });
  });
  describe('#startTrip', () => {
    it('creates the correct action', () => {
      const action = actions.startTrip();
      expect(action).toMatchSnapshot();
    });
  });
  describe('#setSearch', () => {
    it('creates the correct action of searching destinations', () => {
      const action = actions.setSearch('Co');
      expect(action).toMatchSnapshot();
    });
  });
  describe('#setDestination', () => {
    it('creates the correct action of setting the selected destination', () => {
      const action = actions.setDestination(10);
      expect(action).toMatchSnapshot();
    });
  });
  describe('#resetDestination', () => {
    it('creates the correct action of resetting the selected destination', () => {
      const action = actions.resetDestination();
      expect(action).toMatchSnapshot();
    });
  });
});
