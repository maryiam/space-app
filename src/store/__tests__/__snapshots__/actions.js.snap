// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/actions #resetDestination creates the correct action of resetting the selected destination 1`] = `
Object {
  "type": "order/RESET_DESTINATION",
}
`;

exports[`store/actions #setDestination creates the correct action of setting the selected destination 1`] = `
Object {
  "destinationId": 10,
  "type": "order/SET_DESTINATION",
}
`;

exports[`store/actions #setPayment creates the correct action 1`] = `
Object {
  "paymentId": 0,
  "type": "order/SET_PAYMENT",
}
`;

exports[`store/actions #setSearch creates the correct action of searching destinations 1`] = `
Object {
  "searchCriteria": "Co",
  "type": "destination/SET_SEARCH",
}
`;

exports[`store/actions #startTrip creates the correct action 1`] = `
Object {
  "type": "order/SUBMIT",
}
`;
